electrum/fee_policy.py

```diff
@@ -0,0 +1,369 @@
+from typing import Optional, Sequence, Tuple, Union, TYPE_CHECKING
+from decimal import Decimal
+from numbers import Real
+from enum import IntEnum
+
+from .i18n import _
+from .util import NoDynamicFeeEstimates, quantize_feerate, format_fee_satoshis
+from . import util
+from . import constants
+
+if TYPE_CHECKING:
+    from .network import Network
+
+FEE_ETA_TARGETS = [25, 10, 5, 2, 1]
+FEE_DEPTH_TARGETS = [10_000_000, 5_000_000, 2_000_000, 1_000_000,
+                     800_000, 600_000, 400_000, 250_000, 100_000]
+FEERATE_STATIC_VALUES = [1000, 2000, 5000, 10000, 20000, 30000,
+                         50000, 70000, 100000, 150000, 200000, 300000]
+
+# satoshi per kbyte
+FEERATE_MAX_DYNAMIC = 1500000
+FEERATE_WARNING_HIGH_FEE = 600000
+FEERATE_FALLBACK_STATIC_FEE = 150000
+FEERATE_DEFAULT_RELAY = 1000
+FEERATE_MAX_RELAY = 50000
+
+# warn user if fee/amount for on-chain tx is higher than this
+FEE_RATIO_HIGH_WARNING = 0.05
+
+
+def closest_index(value, array) -> int:
+    dist = list(map(lambda x: abs(x - value), array))
+    return min(range(len(dist)), key=dist.__getitem__)
+
+
+class FeeMethod(IntEnum):
+    FIXED = 0
+    STATIC = 1
+    ETA = 2
+    MEMPOOL = 3
+
+
+class FeePolicy:
+    # object associated to a fee slider
+
+    def __init__(self, descriptor: str):
+        try:
+            name, value = descriptor.split(':')
+            self.method = FeeMethod[name.upper()]
+            self.value = int(value)
+        except Exception:
+            # default values
+            self.method = FeeMethod.ETA
+            self.value = 2
+
+    def __repr__(self):
+        return self.get_descriptor()
+
+    def get_descriptor(self) -> str:
+        return self.method.name.lower() + ':' + str(self.value)
+
+    def set_method(self, method: FeeMethod):
+        assert isinstance(method, FeeMethod)
+        self.method = method
+        self.value = 0
```

Comment:
why 0?
A value of 0 means that, in the GUI, when selecting a different FeeMethod, the slider jump to one of the extremes:
- for mempool or ETA, it jumps to a very high feerate (0 block ETA, or 0 MB from tip)
- from static rate, it jumps to a very low feerate (0 s/b)

Maybe we should have sensible defaults for each `FeeMethod` instead, somewhere in the middle of the slider.

Reply:
added more sensible values